src/ImageSharp.Web/ExifOrientationUtilities.cs

```diff
@@ -0,0 +1,212 @@
+            // constant integer size to the transform builder.
+            Vector2 scaled = Scale(position, min, max);
+            AffineTransformBuilder builder = new();
+            Size size = new(1, 1);
+            switch (orientation)
+            {
+                case ExifOrientationMode.TopRight:
+                    builder.AppendTranslation(new Vector2(FlipScaled(scaled.X), 0));
+                    break;
+                case ExifOrientationMode.BottomRight:
+                    builder.AppendRotationDegrees(180);
+                    break;
+                case ExifOrientationMode.BottomLeft:
+                    builder.AppendTranslation(new Vector2(0, FlipScaled(scaled.Y)));
+                    break;
+                case ExifOrientationMode.LeftTop:
+                    builder.AppendTranslation(new Vector2(FlipScaled(scaled.X), 0));
+                    builder.AppendRotationDegrees(270);
+                    break;
+                case ExifOrientationMode.RightTop:
+                    builder.AppendRotationDegrees(270);
+                    break;
+                case ExifOrientationMode.RightBottom:
+                    builder.AppendTranslation(new Vector2(FlipScaled(scaled.X), 0));
+                    builder.AppendRotationDegrees(90);
+                    break;
+                case ExifOrientationMode.LeftBottom:
+                    builder.AppendRotationDegrees(90);
+                    break;
+                default:
+                    return position;
+            }
+
+            Matrix3x2 matrix = builder.BuildMatrix(size);
+            return DeScale(Vector2.Transform(scaled, matrix), SwapXY(min, orientation), SwapXY(max, orientation));
+        }
+
+        /// <summary>
+        /// Transforms the specified size value depending on the specified <paramref name="orientation" />.
+        /// </summary>
+        /// <param name="size">The input size value.</param>
+        /// <param name="orientation">The EXIF orientation.</param>
+        /// <returns>
+        /// The transformed size.
+        /// </returns>
+        public static Size Transform(Size size, ushort orientation)
+            => IsExifOrientationRotated(orientation)
+            ? new Size(size.Height, size.Width)
+            : size;
+
+        /// <summary>
+        /// Transforms the specified anchor value depending on the specified <paramref name="orientation" />.
+        /// </summary>
+        /// <param name="anchor">The input anchor value.</param>
+        /// <param name="orientation">The EXIF orientation.</param>
+        /// <returns>
+        /// The transformed anchor.
+        /// </returns>
+        public static AnchorPositionMode Transform(AnchorPositionMode anchor, ushort orientation)
+        {
+            if (orientation is >= ExifOrientationMode.Unknown and <= ExifOrientationMode.TopLeft)
+            {
+                return anchor;
+            }
```

Comment:
Does this short-circuit really have a performance improvement compared to the switch statements below?

Reply:
Yeah though I should swap the arguments.